### PR TITLE
Feat/esdt relayed arwen

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: process/smartContract/process.go [hunk 1]
```diff
@@ -1327,35 +1342,54 @@ func (sc *scProcessor) createSmartContractResults(
...
 
 	rcvAddress := tx.GetSndAddr()
 	if callType == vmcommon.AsynchronousCallBack {
 		rcvAddress = tx.GetRcvAddr()
 	}
 
+	gasRemaining := vmOutput.GasRemaining
+	var refundGasToRelayerSCR *smartContractResult.SmartContractResult
+	relayedSCR, isRelayed := isRelayedTx(tx)
+	if isRelayed && callType != vmcommon.AsynchronousCall && gasRefund.Cmp(zero) > 0 {
```
Comment (SebastianMarian) [inline process/smartContract/process.go:1366]: You can extract into a variable the whole condition. Ex.: shouldRefundGasToRelayerSCR

File: process/smartContract/process.go [hunk 2]
```diff
@@ -1327,35 +1342,54 @@ func (sc *scProcessor) createSmartContractResults(
...
+		refundGasToRelayerSCR = &smartContractResult.SmartContractResult{
+			Nonce:          relayedSCR.Nonce + 1,
+			Value:          big.NewInt(0).Set(gasRefund),
+			RcvAddr:        relayedSCR.RelayerAddr,
+			SndAddr:        tx.GetRcvAddr(),
+			PrevTxHash:     relayedSCR.OriginalTxHash,
```
Comment (SebastianMarian) [inline process/smartContract/process.go:1372]: Also PrevTxHash should be set to relayedSCR.OriginalTxHash ?

File: process/smartContract/process.go (hunk 1 above)
Comment (sasurobert) [reply process/smartContract/process.go:1488]: done

File: process/smartContract/process.go (hunk 2 above)
Comment (sasurobert) [reply process/smartContract/process.go:1494]: done

**Review by iulianpascalau — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
Initial system tests passed.

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**